src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_ctrl.c

```diff
@@ -556,29 +556,25 @@ static void ert_ctrl_legacy_fini(struct ert_ctrl *ec)
 	return;
 }
 
-static void ert_ctrl_unset_xgq(struct platform_device *pdev)
+static void ert_ctrl_unset_xgq(struct platform_device *pdev, void *xgq_handle)
 {
 	struct ert_ctrl *ec = platform_get_drvdata(pdev);
 	xdev_handle_t xdev = xocl_get_xdev(pdev);
-	int i = 0;
-
-	for (i = 0; i < ec->ec_num_xgq_ips; i++) {
-		struct ert_ctrl_xgq_cu  *xgq_ips = &ec->ec_xgq_ips[i];
+	int xgq_id = xocl_get_xgq_id(xgq_handle);
```

Comment:
xocl_get_xgq_id() cound return -EINVAL. But xgq_id here is not checking that error.

Reply:
Fixed. Please reiew.